src/lib/functions/registry.js

```diff
@@ -113,7 +113,18 @@ class FunctionsRegistry {
     // If there is no watcher for this function but the build produced files,
     // we create a new watcher and watch them.
     if (srcFilesDiff.added.size !== 0) {
-      const newWatcher = await watchDebounced([...srcFilesDiff.added], {
+      const config = func.config.functions
+      const filesToWatch = [
+        ...srcFilesDiff.added,
+        // Include the manually specified files from config. These files might not be
+        // source files, but might still be used/loaded in the function and therefore
+        // we need to watch them to rebuild once they change
+        ...((config && config[func.name] && config[func.name].included_files) || []).filter(
+          (glob) => !glob.startsWith('!'),
+        ),
+        ...((config && config['*'] && config['*'].included_files) || []).filter((glob) => !glob.startsWith('!')),
```

Comment:
These two lines really shouldn't be here, but all the alternatives were worse imho.
Moving the logic to zip-it-ship-it,and adding the globs to the result of the build command, seemed slightly wrong, because the result names the files src files, which these files really aren't necessary.
Adding a new node api to do this in zisi also seems wrong because its api shouldn't do config management.

Ideally there should be a package `@netlify/config` that handles all this. Then cli and zisi (and potentially other packages) could use it.